README.md

```diff
@@ -25,34 +25,69 @@ numpy
 
 [Github link](https://github.com/bytewax/imputing-missing-values)
 
-### Input Code
+### Important Concepts
+
+Bytewax is based around the concept of a **dataflow**. A dataflow is made up of a sequence of operators that interact with data that is “flowing” through it. The dataflow is a directed graph where the nodes are operators and the edges are the data that flows between them. The dataflow is a powerful abstraction that allows you to build complex data processing pipelines with ease.
+
+**Stateless vs stateful** - In Bytewax, operators can be either stateless or stateful. A stateless operator is one that does not maintain any state between invocations. A stateful operator, on the other hand, maintains some state between invocations. This state can be used to store information about the data that has been seen so far, or to store the results of some computation.
+
+**Workers** - A worker is a process that runs a dataflow. Workers are responsible for executing the operators in the dataflow and passing data between them. Workers can run on a single machine, or they can be distributed across multiple machines.
+
+### Goal: generate a dataflow that will impute missing values in a stream of data of random integers and `nan` values.
+
+We can represent our dataflow - called `map_eg` through this diagram, in which the data flows through three key steps:
 
-Bytewax is based around the concept of a dataflow. A dataflow is made up of a sequence of operators that interact with data that is “flowing” through it.
+1. input
+2. stateful map
+3. output
+
+![](./diagram.png)
```

Comment:
Can it be replaced with a mermaid diagram?

Reply:
Asked David - he gave me a method to turn the dataflow into UML format, I then pasted that onto software to generate the diagram
Investigating if there is a way to turn it into a mermaid graph instead

Reply:
David has set up a PR to generate mermaid graphs from the dataflows
I added a temporary mermaid diagram generated with other software